src/printer/tvmscript_printer.cc

```diff
@@ -1220,6 +1300,19 @@ Doc TVMScriptPrinter::PrintPrimFunc(const PrimFunc& primFunc) {
   std::vector<Doc> params;
   for (const auto& param : op->params) {
     var_not_in_headers_.insert(param.get());
+    auto it = op->buffer_map.find(param);
+    // check if this param is a T.handle
+    if (it != op->buffer_map.end()) {
+      // check if this match_buffer has only the first two arguments specified
+      const auto buf = (*it).second;
+      if (IsSimpleBuffer(buf)) {
+        buf_not_in_headers_.insert(buf.get());
+        Doc buf_param_doc;
+        buf_param_doc << buf->name << ": " << MatchBufferDeclaration(buf);
```

Comment:
This is problematic. See the impl of `AllocBuf` and `AllocBufferDeclaration`, I can see at least 
`memo_buf_`, `memo_var_` are not correctly updated if the buffer doesn't go through these two functions. Possible solution is to unify the new behavior into these two functions

Reply:
this is fixed.